From 08448fdbd31494aae0a096b4678fb91cda3c874f Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Date: Fri, 14 Aug 2020 00:34:32 +0300
Subject: [PATCH 239/302] media: i2c: imx296: Fix test pattern selection

The test pattern selector starts at 0, while the menu control uses value
0 to disable the test pattern generator. Fix it.

Signed-off-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
---
 drivers/media/i2c/imx296.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/i2c/imx296.c b/drivers/media/i2c/imx296.c
index 7a54b91a64bb..d353fe89cb24 100644
--- a/drivers/media/i2c/imx296.c
+++ b/drivers/media/i2c/imx296.c
@@ -304,7 +304,7 @@ static int imx296_s_ctrl(struct v4l2_ctrl *ctrl)
 				     IMX296_PGCTRL_REGEN |
 				     IMX296_PGCTRL_THRU |
 				     IMX296_PGCTRL_CLKEN |
-				     IMX296_PGCTRL_MODE(ctrl->val), &ret);
+				     IMX296_PGCTRL_MODE(ctrl->val - 1), &ret);
 		} else {
 			imx296_write(imx, IMX296_PGCTRL, 0, &ret);
 			imx296_write(imx, IMX296_BLKLEVEL, 0x3c, &ret);
-- 
Regards,

Laurent Pinchart

